sync/src/relayer/error.rs

```diff
@@ -2,6 +2,24 @@ use failure::Fail;
 
 #[derive(Debug, Fail, Eq, PartialEq)]
 pub enum Error {
+    #[fail(display = "internal error: {}", _0)]
+    Internal(Internal),
+    #[fail(display = "misbehavior error: {}", _0)]
+    Misbehavior(Misbehavior),
+    #[fail(display = "ignored error: {}", _0)]
+    Ignored(Ignored),
+}
+
+#[derive(Debug, Fail, Eq, PartialEq)]
+pub enum Internal {
+    #[fail(display = "InflightBlocksReachLimit")]
+    InflightBlocksReachLimit,
```

Comment:
It is better to change the category as below:

* move `InflightBlocksReachLimit` into `Ignore` 
* `MissingIndexesEmpty` is not an `Error`. So I don't think define `MissingIndexesEmpty` is a good idea. If we don't define it, then we can remove the definition of `Internal` (after moving `InflightBlocksReachLimit` into `Ignore` )
* rename `Ignore::BlockStored` -> `Ignore::AlreadyStored`

Reply:
> It is better to change the category as below:
> 
> * move `InflightBlocksReachLimit` into `Ignore`

I don't want to move it, because in this case, the node could not accept this compact block just because it's busy, the node does not ignore it deliberately.

> * `MissingIndexesEmpty` is not an `Error`. So I don't think define `MissingIndexesEmpty` is a good idea. If we don't define it, then we can remove the definition of `Internal` (after moving `InflightBlocksReachLimit` into `Ignore` )

ok

> * rename `Ignore::BlockStored` -> `Ignore::AlreadyStored`

ok